cmake/modules/hunter_pkgconfig_export_target.cmake

```diff
@@ -0,0 +1,128 @@
+# ----------------------------------------------------------------------
+# Auto creation of CMake targets from pkg-config
+#
+# Copyright (C) 2017 Alexandre Pretyman. All rights reserved.
+#
+# Looks for a pkgconfig module of a given name and exports a cmake
+# target of the same name
+# ----------------------------------------------------------------------
+
+include(FindPkgConfig)
+
+include(hunter_internal_error)
+include(hunter_status_debug)
+
+function(hunter_pkgconfig_export_target PKG_CONFIG_MODULE)
+  if(TARGET "${PKG_CONFIG_MODULE}")
+    return()
+  endif()
+  pkg_check_modules(${PKG_CONFIG_MODULE} ${PKG_CONFIG_MODULE})
```

Comment:
> This function will find packages from Hunter because it respects CMAKE_PREFIX_PATH variable.

We changed `PKG_CONFIG_LIBDIR` so it would find packages inside the Hunter build tree: 
https://github.com/ruslo/hunter/blob/bfdc19731a5e467674abd7aef9a227678c4176a6/cmake/modules/hunter_finalize.cmake#L85-L88 so I believe this should work, but did it fail in the tests?

> By the way further documentation reading, since CMake 3.6 we have imported targets from the box:

Doh, so it seems like I wasted my time on this? :P 

Maybe use `if(CMAKE_VERSION VERSION_LESS)` as you proposed and export the target under `PkgConfig::` to have the same behavior?